velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -3681,78 +3840,42 @@ TEST_F(SharedArbitrationTest, concurrentArbitration) {
         return false;
       });
 
-  const int numThreads = 30;
-  const int maxNumZombieTasks = 128;
+  const int numThreads = 32;
   std::vector<std::thread> queryThreads;
+  queryThreads.reserve(numThreads);
   for (int i = 0; i < numThreads; ++i) {
     queryThreads.emplace_back([&, i]() {
-      DuckDbQueryRunner duckDbQueryRunner;
-      folly::Random::DefaultGenerator rng;
-      rng.seed(i);
-      while (!stopped) {
-        std::shared_ptr<core::QueryCtx> query;
-        {
-          std::lock_guard<std::mutex> l(mutex);
-          if (queries.empty()) {
-            queries.emplace_back(newQueryCtx());
-          }
-          const int index = folly::Random::rand32() % queries.size();
-          query = queries[index];
-        }
         std::shared_ptr<Task> task;
-        try {
-          task = AssertQueryBuilder(duckDbQueryRunner)
-                     .queryCtx(query)
-                     .plan(PlanBuilder()
-                               .values(vectors)
-                               .addNode([&](std::string id,
-                                            core::PlanNodePtr input) {
-                                 return std::make_shared<FakeMemoryNode>(
-                                     id, input);
-                               })
-                               .planNode())
-                     .assertResults("SELECT * FROM tmp");
-        } catch (const VeloxException& e) {
-          continue;
+        if (i == 0) {
+          // Write task contains aggregate node, which does not support
+          // multithread aggregation type resolver, so make sure it is built in
+          // a single thread.
+          task =
+              makeWriteTask(vectors, newQueryCtx(kMemoryCapacity), numDrivers);
+        } else if ((i % 4) == 0) {
+          task = makeHashJoinTask(
+              vectors, newQueryCtx(kMemoryCapacity), numDrivers);
+        } else if ((i % 4) == 1) {
+          task = makeOrderByTask(
+              vectors, newQueryCtx(kMemoryCapacity), numDrivers);
+        } else if ((i % 4) == 2) {
+          task =
+              makeTopNTask(vectors, newQueryCtx(kMemoryCapacity), numDrivers);
+        } else {
+          task = makeRowNumberTask(
+              vectors, newQueryCtx(kMemoryCapacity), numDrivers);
         }
+
         std::lock_guard<std::mutex> l(mutex);
-        zombieTasks.emplace_back(std::move(task));
-        while (zombieTasks.size() > maxNumZombieTasks) {
-          zombieTasks.pop_front();
+        if ((folly::Random::rand32() % 10) <= 2) {
+          zombieTasks.emplace_back(std::move(task));
         }
```

Comment:
Could you add to abort (not memory pool abort) a random query in followup? And we have seen and fixed a couple of bugs caused by external abort from coordinator in Prestissimo system test.